### PR TITLE
Add co_stacksize to code objects

### LINKED ISSUE #4545: Add `co_stacksize` to code objects.
Opening this separately from #1838 since it also seems like a good first issue.

Currently, code objects are missing the `co_stacksize` attribute:

```python
def foo(a, b): pass
```

In RustPython:

```python
>>>>> foo.__code__.co_stacksize
AttributeError: 'code' object has no attribute 'co_stacksize'
```

While in Python:

```python
>>> foo.__code__.co_stacksize
1
```

looking at the other methods on [`PyCode`](https://github.com/RustPython/RustPython/blob/1fd557d549cd72e7da83608b99e5464273abd4b3/vm/src/builtins/code.rs#L188) should help, the implementation should be similar to those (also related #4502 #4501)

### PR DESCRIPTION
**Pull request by howjmay**
Closes #4545

### CODE REVIEW

**Comment by fanninpm** [comment]
I need to remind myself to update `Lib/test/test_compile.py` to the version from CPython 3.11.

**Comment by DimitrisJim** [comment]
wow, this makes *a lot* of tests pass :smile:

The following tests in `test_compile` can be marked as passing (i.e remove the `# TODO: RUSTPYTHON` comment and the `expectedFailure` decorator from the test function.

 test_async_for (test.test_compile.TestStackSizeStability) ... unexpected success
test_async_for_else (test.test_compile.TestStackSizeStability) ... unexpected success
test_async_with (test.test_compile.TestStackSizeStability) ... unexpected success
test_for (test.test_compile.TestStackSizeStability) ... unexpected success
test_for_break_continue (test.test_compile.TestStackSizeStability) ... unexpected success
test_for_break_continue_inside_async_with_block (test.test_compile.TestStackSizeStability) ... unexpected success
test_for_break_continue_inside_except_block (test.test_compile.TestStackSizeStability) ... unexpected success
test_for_break_continue_inside_finally_block (test.test_compile.TestStackSizeStability) ... unexpected success
test_for_break_continue_inside_try_finally_block (test.test_compile.TestStackSizeStability) ... unexpected success
test_for_break_continue_inside_with_block (test.test_compile.TestStackSizeStability) ... unexpected success
test_for_else (test.test_compile.TestStackSizeStability) ... unexpected success
test_if (test.test_compile.TestStackSizeStability) ... unexpected success
test_if_else (test.test_compile.TestStackSizeStability) ... unexpected success
test_return_inside_async_with_block (test.test_compile.TestStackSizeStability) ... unexpected success
test_return_inside_except_block (test.test_compile.TestStackSizeStability) ... unexpected success
test_return_inside_finally_block (test.test_compile.TestStackSizeStability) ... unexpected success
test_return_inside_try_finally_block (test.test_compile.TestStackSizeStability) ... unexpected success
test_return_inside_with_block (test.test_compile.TestStackSizeStability) ... unexpected success
test_try_except_as (test.test_compile.TestStackSizeStability) ... unexpected success
test_try_except_bare (test.test_compile.TestStackSizeStability) ... unexpected success
test_try_except_qualified (test.test_compile.TestStackSizeStability) ... unexpected success
test_try_finally (test.test_compile.TestStackSizeStability) ... unexpected success
test_while_else (test.test_compile.TestStackSizeStability) ... unexpected success
test_with (test.test_compile.TestStackSizeStability) ... unexpected success

**Comment by DimitrisJim** [comment]
Heya @howjmay, if there's confusion on how to do it or you don't have enough time to do so let me know and I'll fix the tests as needed.

**Comment by howjmay** [comment]
Hi @DimitrisJim I was busy in the past week. I am starting to fix it once again

**Comment by howjmay** [comment]
@DimitrisJim I am feel sorry that I still can't solve this. I have no idea how to get the correct number

**Comment by fanninpm** [comment]
@howjmay try going into `Lib/test/test_compile.py` and removing the lines that say

```python
# TODO: RUSTPYTHON
@unittest.expectedFailure
```

**Comment by howjmay** [comment]
@DimitrisJim Ah thanks! I was totally misunderstood. Let me fix it!

**Comment by DimitrisJim** [comment]
yeah, sorry if I didn't make myself clear enough. The idea here is to remove any decorators (and their preceding comment) for the tests that now pass. If you revert the changes you've made to `test_compile.py` and then run:

```bash
cargo run --release --features ssl,jit -- -m test -j 1 -u all --slowest --fail-env-changed -v test_compile
```

this will execute the test runner for this specific file. After you do that you'll notice that for some of the functions (the ones in https://github.com/RustPython/RustPython/pull/4554#issuecomment-1440526760) the test actually passed. For these you'll just need to remove the `# TODO: RUSTPYTHON` comment and `@unittest.expectedFailure` so the test runner can now accept them as succeeding.

Hope I made it clear! Feel free to ask if something is confusing.

**Comment by howjmay** [comment]
Thank you @DimitrisJim. I just pushed a commit to fix

**Review by DimitrisJim — APPROVED** [review]
thanks @howjmay! Hopefully this now satisfies the test runner.

File: Lib/test/test_compile.py [hunk 1]
```diff
@@ -811,8 +811,6 @@ def test_and_or(self):
     def test_chained_comparison(self):
         self.check_stack_size("x < " * self.N + "x")
 
-    # TODO: RUSTPYTHON
-    @unittest.expectedFailure
```
[the remark this answers is not in the document]
Comment (howjmay) [reply Lib/test/test_compile.py]: done

**Comment by howjmay** [comment]
Thanks!

**Pull request merged**